src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -128,16 +129,187 @@ internal UnitTestResult[] Execute()
             result[result.Length - 1] = newResult;
 #pragma warning restore IDE0056 // Use index operator
         }
+
+        if (_testMethodInfo.Parent.ClassAttribute is STATestClassAttribute
```

Comment:
You may want to check that it inherits from `STATestClass` to leave the opportunity to users to extend the behavior (this will need a specific test).